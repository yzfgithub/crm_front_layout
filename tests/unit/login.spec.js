import Vue from 'vue'
import ElementUI from 'element-ui'

//差 组件识别，异步返回
Vue.use(ElementUI)

import { mount } from '@vue/test-utils'
import Login from '@/packages/login/login.vue'

import axios from 'axios'
// jest.mock('axios', () => ({
//     post: jest.fn(() => Promise.resolve({
//         data: {
//             success: false,
//             info: '用户不存在！'
//         }
//     }))
// }))
// const myMockFn = jest.fn(()=>'difault').mockImplementationOnce(()=>'first call').mockImplementationOnce(()=>'second call')
// console.log(myMockFn,myMockFn,myMockFn,myMockFn)
jest.mock('axios', () => ({
    post: jest.fn()
        .mockImplementationOnce(() => Promise.resolve({
            data: {
                success: false,
                info: '用户不存在！'
            }
        }))
        .mockImplementationOnce(() => Promise.resolve({
            data: {
                success: true,
                token: '123123' // 随意返回一个token
            }
        }))
}))
describe('login',()=>{
    let wrapper;
    beforeEach(()=>{
        wrapper = mount(Login)
    })
    it('Should have the expected html structure',()=>{
        expect(wrapper.element).toMatchSnapshot() //快照测试
    })
    it('failed to login if not typing the correct password',async () =>{
        wrapper.setData({
            account:'yzf',
            password:'123'
        })
        const result = await wrapper.vm.login();
        // expect(result.data.success).toBe(true) //期望返回的数据里success是false  //DATA ERROR !!!WHY
        // expect(result.data.info).toBe('password error')
    })
})

