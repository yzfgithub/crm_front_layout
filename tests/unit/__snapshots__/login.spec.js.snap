// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login Should have the expected html structure 1`] = `
<div
  class="login_bg"
>
  <div
    class="box"
  >
    <div
      class="logo"
    >
      <img
        alt=""
        src="../../assets/logo.png"
      />
    </div>
     
    <div
      class="box-left"
    >
      <img
        alt=""
        src="../../assets/login_left.png"
      />
    </div>
     
    <div
      class="box-right"
    >
      <div
        class="title"
      >
        WLC单点登陆登陆
      </div>
       
      <div
        class="user"
      >
        <span>
          账号
        </span>
         
        <div
          class="input"
        >
          <input
            placeholder="请输入拼音缩写"
            type="text"
          />
        </div>
      </div>
       
      <div
        class="qrcode"
      >
        <div
          class="input"
        >
          <input
            placeholder="请输入拼音缩写"
            type="text"
          />
        </div>
         
        <div
          class="qrcode_btn"
        >
          
                    钉钉验证码
                
        </div>
      </div>
       
      <div
        class="btn"
      >
        
                登录
            
      </div>
    </div>
  </div>
</div>
`;
